File: java/challenge_5/src/test/java/com/ministryoftesting/pageobjects/HomePage.java
package com.ministryoftesting.pageobjects;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.How;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.util.List;

public class HomePage extends BasePage {

    @FindBy(how = How.CSS, using = ".hotel-logoUrl")
    private WebElement imgHotelLogo;

    public HomePage(WebDriver driver) {
        super(driver);

        WebDriverWait wait = new WebDriverWait(driver, 10);
        wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector(".openBooking")));
    }

    public WebElement getHotelLogo(){
        return imgHotelLogo;
    }

}

File: dotnet/MoTAutomationWeekDotNet/Challenge_5/PageObjects/HomePage.cs
using OpenQA.Selenium;
using OpenQA.Selenium.Support.UI;
using System;

namespace Challenge_5.PageObjects
{
    public class HomePage : BasePage
    {

        private IWebElement imgHotelLogo
        {
            get { return Driver.FindElement(By.CssSelector(".hotel-logoUrl")); }
        }

        public HomePage(IWebDriver driver) : base(driver)
        {
            
            WebDriverWait wait = new WebDriverWait(driver, TimeSpan.FromSeconds(10));
            wait.Until(driver =>
            {
                return driver.FindElement(By.CssSelector(".openBooking")).Displayed;
            });
        }

        public IWebElement GetHotelLogo()
        {
            return imgHotelLogo;
        }
    }
}
